src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -100,18 +100,28 @@ Optional<ExternalObjectDirectoryEntity> findMatchingExternalObjectDirectoryEntit
             WHERE eod.status in :failedStatuses
             AND eod.externalLocationType = :type
             AND eod.transferAttempts <= :transferAttempts
+            AND eod.osrUuid is null
```

Comment:
this is interesting. I wonder whether some of the other repo methods would need a similar change applied to make sure they filter out the DETS EOD records.
@hemantasharma1129 ?